### ISSUE TITLE
Failed to get subtrees request running datastore

### ISSUE THREAD

**Issue by AJR-Dev0015** [sysrepo/sysrepo#2567]
Hello

I am having trouble in my sysrepo datastore.

I installed my own YANG module to test if datastore of sysrepo can return information based on the request.

Module Name                | Revision   | Conformance | Data Owner          | Permissions | Submodules                    | Enabled Features
-----------------------------------------------------------------------------------------------------------------------------------------------
ietf-netconf-notifications | 2012-02-06 | Installed   | root:root           | 666         |                               |
ietf-netconf               | 2011-06-01 | Imported    |                     |             |                               |
ietf-netconf-acm           | 2018-02-14 | Imported    |                     |             |                               |
nc-notifications           | 2008-07-14 | Installed   | root:root           | 666         |                               |
notifications              | 2008-07-14 | Installed   | root:root           | 666         |                               |
ultraconfig-interfaces     | 2020-01-03 | Installed   | root:root           | 666         |                               |
ietf-interfaces            | 2018-02-20 | Installed   | root:root           | 666         |                               |
ietf-ip                    | 2018-02-22 | Installed   |                     |             |                               |
building                   | 2021-09-06 | Installed   | ubuntu:ubuntu       | 664         |                               |
filter-ietf-interfaces     | 2021-09-07 | Installed   | root:root           | 666         |

<?xml version='1.0' encoding='UTF-8'?>  
<interfaces xmlns="urn:ietf:params:xml:ns:yang:filter-ietf-interfaces">
  <interface>
    <name>GigabitEthernet2</name>
  </interface>
</interfaces>
<interfaces-state xmlns="urn:ietf:params:xml:ns:yang:filter-ietf-interfaces">
  <interface>
    <name>GigabitEthernet2</name>
  </interface>
</interfaces-state>

When I tried to use my gNMI client using GET only using  PATH = ["filter-ietf-interfaces:interfaces"]

I get this error always:
[DBG] Processing get_subtrees request.
[INF] Get subtrees request running datastore, xpath: /filter-ietf-interfaces:interfaces
[DBG] Module filter-ietf-interfaces already loaded
[ERR] Get subtrees failed for xpath '/filter-ietf-interfaces:interfaces'

In my sysrepo directory I have this

ubuntu@ub1:~/gnmi_client$ ll /etc/sysrepo/data/
total 28
drwxr-xr-x 4 root   root   4096 Sep  8 01:16 ./
drwxr-xr-x 4 root   root   4096 Aug 28 00:20 ../
-rw-rw-r-- 1 ubuntu ubuntu    0 Sep  7 03:16 building.persist
-rw-rw-r-- 1 ubuntu ubuntu    0 Sep  7 03:16 building.running
-rw-rw-r-- 1 ubuntu ubuntu    0 Sep  7 03:16 building.running.lock
-rw-rw-r-- 1 ubuntu ubuntu   55 Sep  7 21:11 building.startup
-rw-rw-r-- 1 ubuntu ubuntu    0 Sep  7 03:16 building.startup.lock
-rw-rw-rw- 1 root   root      0 Sep  8 01:16 filter-ietf-interfaces.persist
-rw-rw-rw- 1 root   root      0 Sep  8 01:16 filter-ietf-interfaces.running
-rw-rw-rw- 1 root   root      0 Sep  8 01:16 filter-ietf-interfaces.running.lock
-rw-rw-rw- 1 root   root    143 Sep  8 01:18 filter-ietf-interfaces.startup
-rw-rw-rw- 1 root   root      0 Sep  8 01:16 filter-ietf-interfaces.startup.lock

Do you have a suggestion or maybe I missed the part of the installation.

**Comment by michalvasko** [comment]
Based on your *sysrepoctl* output and the `lock` datastore files, I assume you are using the legacy version of *sysrepo* (0.7.x). I do not know why the XPath filter does not work and since we do not provide support for this version anymore, I cannot help you. Please update to a more recent version of *sysrepo*, there are 2 more generations (v1 and v2) available.

**Comment by AJR-Dev0015** [comment]
Thank you very much.

Can you provide me the link so that I can correctly get the version that you are referring?

**Comment by michalvasko** [comment]
To get full support, you should be using the latest version, which is kept in the `devel` branch ((link)[https://github.com/sysrepo/sysrepo/tree/devel]). But note that you also need to have `devel` branches installed from all the dependency projects (*libyang*, if using *netopeer2* then also *libnetconf2*).

**Comment by AJR-Dev0015** [comment]
Hello.

I am very sorry.

I have tried using the link(https://github.com/sysrepo/sysrepo/tree/devel)
you provided and installed sysrepo.


before installation of sysrepo, I installed libyang(https://github.com/CESNET/libyang/tree/devel)

During the  make build, I encountered many errors in the compilation.

Please see below image for some of the compilation error.


Do you have any suggestion about the error compilation?

**Comment by michalvasko** [comment]
Yes, you are still compiling against the old *libyang*. Remove *libyang* completely, install the new version, and then try again. Starting with an empty `build` directory in *sysrepo*.

**Comment by AJR-Dev0015** [comment]
Hello.

Thank you very much for checking on it.

I am trying to install "https://github.com/YohanPipereau/sysrepo-gnxi"

It seems  in the procedure with the link "https://github.com/YohanPipereau/sysrepo-gnxi/releases"

it mentioned about installing the .deb file.

The .deb file "sysrepo_0.7.7-1_amd64.deb" seems to be the cause of the problem.
The content of this is the old version x.7.x.
This seems to be cause of the problem.


I would just like to clarify my understanding about sysrepo.

When will "running" data store will have a data?

When using get_items, where does the data will be retrieved?

**Comment by michalvasko** [comment]
> I am trying to install "https://github.com/YohanPipereau/sysrepo-gnxi"

Then you should probably ask the author for help if they recommend using *sysrepo* 0.7.7, we do not support it anymore.

> When will "running" data store will have a data?

When you store some data in this datastore. You can use `sysrepocfg` tool for that, for example. It allows all the operations on all the datastores, to configure data you can run `sysrepocfg -Evim -d running -f xml -m <YANG-module>` to edit the module running data using `vim` (you can specify other editors). XML format is expected, you can also use JSON.

**Comment by AJR-Dev0015** [comment]
Thank you very much for the insights.

> When you store some data in this datastore. You can use sysrepocfg tool for that, for example. It allows all the operations on > all the datastores, to configure data you can run sysrepocfg -Evim -d running -f xml -m <YANG-module> to edit the module > running data using vim (you can specify other editors). XML format is expected, you can also use JSON.

I can edit the YANG-module provided that it has data right?
(sysrepocfg --editor=nano -d running -f xml --module=filter-ietf-interfaces)

I am trying to edit the module but  Module name is not specified is outputted.

**Comment by michalvasko** [comment]
Oh, so you continue to use *sysrepo* 0.7.7? Then try `sysrepocfg -h`, it has different syntax than the current version. Other than that you are on your own, sorry.

**Comment by AJR-Dev0015** [comment]
Hello

> Oh, so you continue to use _sysrepo_ 0.7.7? Then try `sysrepocfg -h`, it has different syntax than the current version. Other than that you are on your own, sorry.

I am very sorry for that.
I am just trying to understand the concept of the current sysrepo that is why I did not migrated it yet.

Thank you anyway for the time.

I will try to install to new version.
